src/components/apps/listing/TableView.js

```diff
@@ -0,0 +1,250 @@
+    TableRow,
+} from "@material-ui/core";
+import ids from "../ids";
+import TableLoading from "../../utils/TableLoading";
+import AppStatusIcon from "../AppStatusIcon";
+import AppName from "../AppName";
+import AppFields from "../AppFields";
+import { injectIntl } from "react-intl";
+import messages from "../messages";
+
+function getTableColumns(deletable, enableMenu) {
+    let tableColumns = [
+        {
+            name: "",
+            numeric: false,
+            enableSorting: false,
+            key: AppFields.STATUS.key,
+            id: AppFields.STATUS.key,
+        },
+        {
+            name: AppFields.NAME.fieldName,
+            enableSorting: true,
+            key: AppFields.NAME.key,
+            id: AppFields.NAME.key,
+        },
+        {
+            name: AppFields.INTEGRATOR.fieldName,
+            enableSorting: true,
+            key: AppFields.INTEGRATOR.key,
+            id: AppFields.INTEGRATOR.key,
+        },
+        {
+            name: AppFields.RATING.fieldName,
+            enableSorting: true,
+            key: AppFields.RATING.key,
+            id: AppFields.RATING.key,
+        },
+        {
+            name: AppFields.SYSTEM.fieldName,
+            enableSorting: false,
+            key: AppFields.SYSTEM.key,
+            id: AppFields.SYSTEM.key,
+            align: "right",
+        },
+    ];
+
+    if (deletable) {
+        tableColumns.push({
+            name: "",
+            enableSorting: false,
+            key: "remove",
+        });
+    }
+
+    if (enableMenu) {
+        tableColumns.push({
+            name: "",
+            enableSorting: false,
+            key: "menu",
+            align: "right",
+        });
+    }
+
+    return tableColumns;
+}
+
+function TableView(props) {
+    const {
+        loading,
+        error,
+        listing,
+        baseId,
+        handleRequestSort,
+        handleSelectAllClick,
+        handleClick,
+        order,
+        orderBy,
+        selected,
+        intl,
+    } = props;
+    const apps = listing?.apps;
+    const columnData = getTableColumns(false, false);
+    const tableId = build(baseId, ids.LISTING_TABLE);
+    return (
+        <TableContainer
+            component={Paper}
+            style={{
+                height: "60vh",
+            }}
+        >
+            <Table stickyHeader={true} size="small">
+                <EnhancedTableHead
+                    selectable={true}
+                    numSelected={selected.length}
+                    rowsInPage={listing?.apps.length}
+                    order={order}
+                    orderBy={orderBy}
+                    baseId={tableId}
+                    columnData={columnData}
+                    onRequestSort={handleRequestSort}
+                    onSelectAllClick={handleSelectAllClick}
+                />
+                {loading && (
+                    <TableBody>
+                        <TableLoading numColumns={6} numRows={25} />
```

Comment:
Should this also use `columnData.length`?
```suggestion
                        <TableLoading numColumns={columnData.length} numRows={25} />
```